src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md

```diff
@@ -0,0 +1,322 @@
+
+## Step 1: Create a new module
+
+Create a new module named `Learning/CustomField` and register it.
+
+For more information about creating a new module, refer to [Create a Module](https://devdocs.magento.com/videos/fundamentals/create-a-new-module/)
+
+## Step 2 Add a `db_schema.xml` file
+
+Add the `paymentpocomment` column in the `quote_payment` and `sales_order_payment` table using the declarative schema method.
+
+Create the declarative schema in the following path `app/code/Learning/CustomField/etc/db_schema.xml`:
+
+{% collapsible Show code %}
+
+```xml
+<schema xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:framework:Setup/Declaration/Schema/etc/schema.xsd">
+    <table name="quote_payment" resource="checkout" engine="innodb" comment="Sales Flat Quote Payment">
+        <column xsi:type="text" name="paymentpocomment" nullable="true" comment="PO Comment"/>
+    </table>
+    <table name="sales_order_payment" resource="sales" engine="innodb" comment="Sales Flat Order Payment">
+        <column xsi:type="text" name="paymentpocomment" nullable="true" comment="PO Comment"/>
+    </table>
+</schema>
+```
+
+## Step 3: Add a requirejs file to the module
+
+Next, create the `requirejs-config.js` file:
+
+`app/code/Learning/CustomField/view/frontend/requirejs-config.js`
+
+```js
+ var config = {
+    map: {
+        '*': {
+          'Magento_OfflinePayments/js/view/payment/offline-payments':'Learning_CustomField/js/view/payment/offline-payments',
+        }
+    }
+}
+```
+
+{% endcollapsible %}
+
+## Step 4: Override the vendor files
+
+Next, override the `offline-payments.js` file to change the renderer of the Purchase Order payment method.
+
+`app/code/Learning/CustomField/view/frontend/web/js/view/payment/offline-payments.js`
+
+{% collapsible Show code %}
+
+```js
+define(
+    [
+        'uiComponent',
+        'Magento_Checkout/js/model/payment/renderer-list'
+    ],
+    function (
+        Component,
+        rendererList
+    ) {
+        'use strict';
+        rendererList.push(
+            {
+                type: 'checkmo',
+                component: 'Magento_OfflinePayments/js/view/payment/method-renderer/checkmo-method'
+            },
+            {
+                type: 'banktransfer',
+                component: 'Magento_OfflinePayments/js/view/payment/method-renderer/banktransfer-method'
+            },
+            {
+                type: 'cashondelivery',
+                component: 'Magento_OfflinePayments/js/view/payment/method-renderer/cashondelivery-method'
+            },
+            {
+                type: 'purchaseorder',
+                component: 'Learning_CustomField/js/view/payment/method-renderer/purchaseorder-method'
+            }
+        );
+        /** Add view logic here if needed */
+        return Component.extend({});
+    }
+);
+```
+
+{% endcollapsible %}
+
+It is also necessary to override the `purchaseorder-method.js` file.
```

Comment:
Explain why this file needs to be overridden

Reply:
Provided an explanation.